src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,117 @@
+#include <aikido/constraint/SequentialSampleable.hpp>
+
+#include <dart/common/StlHelpers.hpp>
+
+namespace aikido {
+namespace constraint {
+
+// TODO (avk): Should I declare the class first and then define?
+// What is "good" C++ convention? Don't tell me it's a preference JS! -.-
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>> generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(std::move(generators))
+    , mIndex(0)
+  {
+    // Do nothing
+    // TODO (avk): Should we again check equivalence of statespaces between
+    // that of each generator and mStateSpace? I don't think it's necessary.
+  }
+
+  statespace::StateSpacePtr getStateSpace() const override
+  {
+    return mStateSpace;
+  }
+
+  bool sample(statespace::StateSpace::State* state)
+  {
+    if (!mGenerators[mIndex]->canSample())
+      mIndex++;
```

Comment:
I think this function should keep iterating until it finds a sampleable generator instead of just check only one generator.